Lib/multiprocessing/reduction.py

```diff
@@ -68,12 +68,19 @@ def dump(obj, file, protocol=None):
     __all__ += ['DupHandle', 'duplicate', 'steal_handle']
     import _winapi
 
-    def duplicate(handle, target_process=None, inheritable=False):
+    def duplicate(handle, target_process=None, inheritable=False,
+                  *, source_pid=None):
         '''Duplicate a handle.  (target_process is a handle not a pid!)'''
+        current_process = _winapi.GetCurrentProcess()
+        if source_pid is not None:
+            source_process_handle = _winapi.OpenProcess(
+                _winapi.PROCESS_DUP_HANDLE, False, source_pid)
+        else:
+            source_process_handle = current_process
         if target_process is None:
-            target_process = _winapi.GetCurrentProcess()
+            target_process = current_process
         return _winapi.DuplicateHandle(
-            _winapi.GetCurrentProcess(), handle, target_process,
+            source_process_handle, handle, target_process,
             0, inheritable, _winapi.DUPLICATE_SAME_ACCESS)
 
     def steal_handle(source_pid, handle):
```

Comment:
So is this function still used?

Reply:
It's part of the public API, no? At least, it's exported in __all__ on Windows.